### PR TITLE
Ecomm 5 crud base

### PR DESCRIPTION
**Pull request by BrianVega**

### CODE REVIEW

File: src/main/java/org/ecommerce/util/CrudOperations.java [hunk 1]
```diff
@@ -3,10 +3,8 @@
 import java.util.List;
 
 public interface CrudOperations<T> {
```
Comment (algutierrez-griddynamics) [inline src/main/java/org/ecommerce/util/CrudOperations.java:5]: Please use generics for the id data type

File: src/main/java/org/ecommerce/Main.java [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
-    private static final Map<String, Object> REQUEST = new HashMap<>();
-    private static final CustomerRepository customerRepository = new CustomerRepository();
-    private static final UserController customerController = new CustomerController(new CustomerService(customerRepository), new PasswordServiceImpl(customerRepository));
+    private static final Map<String, String> request = new HashMap<>();
+    private static final UserRepository customerRepository = new UserRepository();
+    private static final PasswordService passwordService = new PasswordServiceImpl(customerRepository);
+    private static final EmailService emailService = new EmailServiceImpl();
+    private static final UserServiceImpl customerService = new UserServiceImpl(customerRepository, passwordService, emailService);
+
+    private static final UserController customerController = new UserController(customerService, passwordService);
```
Comment (algutierrez-griddynamics) [inline src/main/java/org/ecommerce/Main.java:21]: Move this logic to test class

File: pom.xml [hunk 3]
```diff
@@ -15,6 +15,11 @@
     </properties>
 
     <dependencies>
+        <dependency>
+            <groupId>org.springframework</groupId>
```
Comment (algutierrez-griddynamics) [inline pom.xml:19]: Why is this needed?

Comment (BrianVega) [reply pom.xml]: Actually it is not, at least not yet, i just deleted it and updated the POM https://github.com/algutierrez-griddynamics/e-commerce/pull/11/commits/3639dd11913cad56656ac7f5bc4b8d524ad40841, i also added some testing dependencies as well as some plugins https://github.com/algutierrez-griddynamics/e-commerce/pull/11/commits/8d58e474e91d7c02ae0296a744c2733d03d37ee8 and lombok https://github.com/algutierrez-griddynamics/e-commerce/pull/11/commits/1ff7a447bda373a2ad6a8786e839c97763e2b17c

File: src/main/java/org/ecommerce/util/CrudOperations.java (hunk 1 above)
Comment (BrianVega) [reply src/main/java/org/ecommerce/util/CrudOperations.java]: https://github.com/algutierrez-griddynamics/e-commerce/pull/11/commits/01e81bdbd914d57659ee275a2dcc76bea1c49f4c done

File: src/main/java/org/ecommerce/Main.java (hunk 2 above)
Comment (BrianVega) [reply src/main/java/org/ecommerce/Main.java]: I just cleared the main method https://github.com/algutierrez-griddynamics/e-commerce/pull/11/commits/a1ea6b262c307306d708b8e3ca63722513c6ddd3

File: pom.xml [hunk 4]
```diff
@@ -12,13 +12,220 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-junit-jupiter</artifactId>
+            <version>${mockito-version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
```
Comment (algutierrez-griddynamics) [inline pom.xml:84]: when is Lombok used?
